### PR TITLE
Docs: Installing on TrueNAS

### PR DESCRIPTION
**Pull request by DjP-iX**
This PR introduces documentation on installing the Jellyfin application in TrueNAS SCALE. 

Note: I've left the `sidebar_position` undefined to allow you to weight as needed.

### CODE REVIEW

File: docs/general/installation/truenas.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+For **Jellyfin Transcodes Storage**, choose **ixVolume**, **Host Path**, or **emptyDir (Temporary directory created on the disk or in memory)**. An emptyDir uses ephemeral storage either on the disk or by mounting a tmpfs (RAM-backed filesystem) directory for storing transcode files.
+
+#### Mounting Additional Storage
+
+Click **Add** next to **Additional Storage** to add the media storage path(s) on your system.
+
+![Additional Storage](/images/docs/install-truenas-11.png)
+
+Select **iXvolume (dataset created automatically by the system)**, **Host Path (Path that already exists on the system)**, or **SMB Share (Mounts a persistent volume claim to a SMB share)** in **Type**.
```
[the remark this answers is not in the document]
Comment (DjP-iX) [reply docs/general/installation/truenas.md:148]: That is actually a known issue in TN 23.10.0 It is resolved in the 23.20.1 version, releasing tomorrow: https://ixsystems.atlassian.net/browse/NAS-124947

**Comment by DjP-iX** [comment]
ty for the suggestions @felix920506! Might take me a few weeks to circle back to this, but I will be reworking to address them

File: docs/general/installation/truenas.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+title: TrueNAS SCALE
+description: Install on TrueNAS SCALE.
+sidebar_position: 7
+---
+
+Jellyfin can easily be installed on [TrueNAS SCALE](https://www.truenas.com/truenas-scale/) (version 23.10.1 or newer) via the **Community** train application. Consider reviewing the [TrueNAS Apps tutorial](https://www.truenas.com/docs/scale/scaletutorials/apps/) if you have not previously configured applications on your system.
+
+TrueNAS SCALE makes installing Jellyfin easy, but you must use the Jellyfin web portal to configure accounts and manage libraries.
+
+## First Steps
```
Comment (felix920506) [inline docs/general/installation/truenas.md:12]:
```suggestion
## Introduction and Preparation
```

File: docs/general/installation/truenas.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+Application configuration settings are presented in several sections, each explained below.
+To find specific fields click in the **Search Input Fields** search field, scroll down to a particular section or click on the section heading on the navigation area in the upper-right corner.
+
+![Install Jellyfin Screen](/images/docs/install-truenas-3.png)
+
+Accept the default values in **Application Name** and **Version**.
+
+Accept the defaults in **Jellyfin Configuration**, **User and Group Configuration**, and **Network Configuration** or change to suit your use case.
+You must select **Host Network** under **Network Configuration** if using [DLNA](https://jellyfin.org/docs/general/networking/dlna/).
```
Comment (felix920506) [inline docs/general/installation/truenas.md:49]:
```suggestion
You must select **Host Network** under **Network Configuration** if using [DLNA](/docs/general/networking/dlna/).
```
image
This message in the TrueNas app store should also be changed if possible

File: docs/general/installation/truenas.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+Jellyfin requires three app storage datasets.
+You can allow SCALE to create them for you, or use the dataset(s) created in [First Steps](#first-steps).
+Select the storage options you want to use for **Jellyfin Config Storage** and **Jellyfin Cache Storage**.
+Select **ixVolume (dataset created automatically by the system)** in **Type** to let SCALE create the dataset or select **Host Path** to use the existing datasets created on the system.
```
Comment (felix920506) [inline docs/general/installation/truenas.md:54]: image
This section should be updated to match reality.

It would also be helpful to give hints on what types of storage this should be on.
For Cache and Config, these should preferably be on solid state storage, and shouldn't be on the same device as the media library if spinning storage is used.
For Transcodes, CMR HDDs can work.

File: docs/general/installation/truenas.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+Jellyfin also requires a dataset or [emptyDir](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) for **Jellyfin Transcodes Storage**.
+Select **ixVolume (dataset created automatically by the system)** in **Type** to let SCALE create the dataset, select **Host Path** to use an existing dataset created on the system, or select **emptyDir** to use a temporary storage volume on the disk or in memory.
+
+Mount one or more media libraries using **Additional Storage**.
+Click **Add** to enter the path(s) on your system.
+Select **Host Path (Path that already exists on the system)** or **SMB Share (Mounts a persistent volume claim to a SMB share)** in **Type**.
+Enter a **Mount Path** to be used within the Jellyfin container. For example, the local Host Path /mnt/tank/video/movies could be assigned the Mount Path /media/movies.
+Define the **Host Path** or complete the SMB Share Configuration fields.
+See [Mounting Additional Storage](#mounting-additional-storage) below for more information.
```
Comment (felix920506) [inline docs/general/installation/truenas.md:64]: I find this section on mounts quite hard to follow, maybe make this like a table or something?
Or maybe just move the section on mounts below to here?

File: docs/general/installation/truenas.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+Accept the defaults in **Resource Configuration** or change the CPU and memory limits to suit your use case.
```
Comment (felix920506) [inline docs/general/installation/truenas.md:66]: I think the resources configuration section at the very bottom of this page would be more suited here.

File: docs/general/installation/truenas.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+### Resource Configuration Settings
+
+Accept the default values in **Resources Configuration** or enter new CPU and memory values
+By default, this application is limited to use no more than 4 CPU cores and 8 gibibytes available memory.
+
+![Resource Limits](/images/docs/install-truenas-12.png)
+
+To customize the CPU and memory allocated to the container Jellyfin uses, enter new CPU values as a plain integer value followed by the suffix m (milli).
+Default is 4000m.
```
Comment (felix920506) [inline docs/general/installation/truenas.md:234]:
```suggestion
Default is 4000m, which means Jellyfin is allowed to use 4 CPU threads by default.
```

File: docs/general/installation/truenas.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+Accept the default value 8Gi allocated memory or enter a new limit in bytes.
+Enter a plain integer followed by the measurement suffix, for example 129M or 123Mi.
```
Comment (felix920506) [inline docs/general/installation/truenas.md:237]: I'd think it would be better to give examples in the GB range. It's not really recommended to allocate less than 4GB of RAM for JF

File: docs/general/installation/truenas.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+
+Systems with compatible GPU(s) for hardware acceleration display devices in **GPU Configuration**.
+Use the **GPU Resource** dropdown menu(s) to configure device allocation.
+
+To isolate a GPU, go to the **System Settings > Advanced** screen and click **Configure** on the **Isolated GPU Device(s)** widget to open the **Isolate GPU PCI’s Ids** screen. Select the GPU device ID to isolate from the dropdown list and click **Save**.
+You must have at least two GPUs in your system; one allocated to the host system for system functions and the other available for use by either a VM or applications.
```
Comment (felix920506) [inline docs/general/installation/truenas.md:243]: Is having a single SR-IOV capable GPU acceptable for this? If so it should be noted, as many Intel Integrated Graphics devices have this capability.

**Review by felix920506 — CHANGES_REQUESTED** [review]
Reading through this guide, I find quite a bit of duplicate information on the bottom. I don't know if this is intended or not. However, I find the info on the bottom more useful when trying to configure the App compared to their counterparts on the top.

There are also a few things, please check my comments in the middle.

File: docs/general/installation/truenas.md (hunk 3 above)
Comment (DjP-iX) [reply docs/general/installation/truenas.md:49]: I believe that tooltip is universal for all apps, not Jellyfin specific, but I will pass that concern along

File: docs/general/installation/truenas.md (hunk 5 above)
Comment (DjP-iX) [reply docs/general/installation/truenas.md:64]: I'm fine with collapsing the two sections, The idea behind the template this is written to is that you have basic install procedure and then more detailed info below, but there's no reason it can't be all in-line

File: docs/general/installation/truenas.md (hunk 7 above)
Comment (DjP-iX) [inline docs/general/installation/truenas.md:234]:
```suggestion
Default is 4000m, which means Jellyfin is allowed to use 4 CPU threads.
```

File: docs/general/installation/truenas.md (hunk 9 above)
Comment (DjP-iX) [reply docs/general/installation/truenas.md:243]: I'm not entirely clear on the question here, but my understanding is a dedicated GPU is required for system processes. An additional SR-IOV capable GPU could be split across multiple apps, but not between system tasks and app processing. Does that make sense?

File: docs/general/installation/truenas.md (hunk 4 above)
Comment (DjP-iX) [reply docs/general/installation/truenas.md:54]: Will add in the storage type recommendations. 

I'm not sure I follow what about this you think does not match reality though?

File: docs/general/installation/truenas.md (hunk 9 above)
Comment (felix920506) [reply docs/general/installation/truenas.md:196]: The question here is, if I only have a single GPU on the system that is capable of SR-IOV, can I use it for both the host and apps?

Comment (DjP-iX) [reply docs/general/installation/truenas.md:196]: It appears I was misinformed here. On further investigation the limitation is VM specific, so a single GPU cannot be split between host and VM system tasks, but a single GPU should work for system and apps. I'm going to suggest simply removing lines 243 and 244.

**Comment by felix920506** [comment]
@DjP-iX is this ready for re-review?

**Comment by DjP-iX** [comment]
I haven't looked at it in some time, but as far as I know it is

File: docs/general/installation/truenas.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+## Finalizing Install
+
+Click **Install**.
+
+A container launches with root privileges to apply the correct permissions to the Jellyfin directories.
+Afterward, the Jellyfin container runs as a non-root user (default: 568).
+Configured storage directory ownership is changed if the parent directory does not match the configured user.
+
+The system opens the **Installed Applications** screen with the Jellyfin app in the **Deploying** state.
+When the installation completes it changes to **Running**.
```
Comment (Gauvino) [inline docs/general/installation/truenas.md:206]:
```suggestion
When the installation completes, it changes to **Running**.
```

**Review by Gauvino — CHANGES_REQUESTED** [review]

File: docs/general/installation/truenas.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+
+You can configure environment variables at any time after deploying the application.
+
+SCALE does not need advance preparation.
+
+You can allow SCALE to create the datasets Jellyfin requires automatically during app installation.
+Or before beginning app installation, [create the datasets](https://www.truenas.com/docs/scale/scaletutorials/storage/datasets/datasetsscale/) to use in the **Storage Configuration** section during installation.
+Jellyfin requires two datasets: **config** and **cache**.
+You can organize these as one parent with two child datasets, for example mnt/tank/jellyfin/config, mnt/tank/jellyfin/cache, and so on.
+You can choose to create a static **transcodes** dataset or use temporary storage in the disk or memory for transcoding.
```
[the remark this answers is not in the document]
Comment (DjP-iX) [reply docs/general/installation/truenas.md:25]:
```suggestion
You can choose to create a static **transcodes** dataset or use temporary storage on the disk or in memory for transcoding.
```

File: docs/general/installation/truenas.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+
+Click **Add** next to **Additional Storage** to add the media library storage path(s) on your system.
+
+![Additional Storage](/images/docs/install-truenas-11.png)
+
+Select **Host Path (Path that already exists on the system)** or **SMB Share (Mounts a persistent volume claim to a SMB share)** in **Type**.
+You can select **iXvolume (dataset created automatically by the system)** to create a new library dataset, but this is not recommended.
+
+Mounting an SMB share allows data synchronization between the share and the app.
+The SMB share mount does not include ACL protections at this time. Permissions are currently limited to the permissions of the user that mounted the share. Alternate data streams (metadata), finder colors tags, previews, resource forks, and MacOS metadata is stripped from the share along with filesystem permissions, but this functionality is undergoing active development and implementation planned for a future TrueNAS SCALE release.
```
[the remark this answers is not in the document]
Comment (DjP-iX) [reply docs/general/installation/truenas.md:113]:
```suggestion
The SMB share mount does not include ACL protections at this time. Permissions are currently limited to the permissions of the user that mounted the share. Alternate data streams (metadata), finder colors tags, previews, resource forks, and MacOS metadata are stripped from the share along with filesystem permissions, but this functionality is undergoing active development and implementation planned for a future TrueNAS SCALE release.
```

File: docs/general/installation/truenas.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+        <td>The quota size for the share volume. You can edit the size after deploying the application if you need to increase the storage volume capacity for the share.</td>
+      </tr>
+    </tbody>
+  </table>
+</details>
+
+### Resource Configuration Settings
+
+Accept the default values in **Resources Configuration** or enter new CPU and memory values
+By default, this application is limited to use no more than 4 CPU cores and 8 gibibytes available memory.
```
[the remark this answers is not in the document]
Comment (DjP-iX) [reply docs/general/installation/truenas.md:182]: This should indeed be gibibytes 

image

File: docs/general/installation/truenas.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Gauvino) [reply docs/general/installation/truenas.md:25]: LGTM

File: docs/general/installation/truenas.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (Gauvino) [reply docs/general/installation/truenas.md:182]: Updated review !

**Review by Gauvino — APPROVED** [review]
LGTM !

**Comment by maru801** [comment]
I submitted a PR to your fork of this to update it with the release of Electric Eel for SCALE.
Hopefully it comes of use.

**Comment by joshuaboniface** [comment]
Superseded by #1228 and credited as coauthor in merge commit.
